nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -575,6 +583,38 @@ in {
       path = with pkgs; [ file imagemagick ffmpeg ];
     };
 
+
+  systemd.services.mastodon-media-prune =
+    let
+      cfg = config.services.mastodon;
+    in
+    lib.mkIf cfg.mediaAutoRemove {
+      description = "Mastodon media prune";
+      environment = lib.filterAttrs (n: _: n != "PATH") config.systemd.services.mastodon-web.environment;
+      serviceConfig = {
+        Type = "oneshot";
+        # Remove remote media attachments and preview cards older than one month.
+        ExecStart = ''
+          ${cfg.package}/bin/tootctl media remove --days=30
+          ${cfg.package}/bin/tootctl preview_cards remove --days=30
+        '';
+        User = cfg.user;
+        Group = cfg.group;
+        EnvironmentFile = "/var/lib/mastodon/.secrets_env";
+        PrivateTmp = true;
+      };
+    };
+
+  systemd.timers.mastodon-media-prune = lib.mkIf cfg.mediaAutoRemove {
```

Comment:
Instead of adding a timer explicitly, you could just set `startAt = "daily";` in the service above. Since you left `AccuracySec` to the default value this should be equivalent.

Reply:
I don't know what the difference is and couldn't find documentation about it.

Reply:
Just heard about it the first time, looks like a neat NixOS shorthand for creating the timer: https://github.com/NixOS/nixpkgs/blob/951f81c0cc39cd0e8fa2575d708aa5eadd3e1dce/nixos/lib/systemd-unit-options.nix#L459

Reply:
Documentation is here: https://nixos.org/manual/nixos/stable/options.html#opt-systemd.services._name_.startAt

In this particular case there's no difference, it's just less code.

Reply:
Ah, I looked for a systemd option of that name :sweat_smile: